services/provider/server/server.go

```diff
@@ -341,77 +345,314 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
+				"csi.storage.k8s.io/fstype": "ext4",
+				"imageFormat":               "2",
+				"mounter":                   "rbd",
+				"mapOptions":                "krbd:rxbounce",
+				"csi.storage.k8s.io/provisioner-secret-name":            rbdProvisionerSecretName,
+				"csi.storage.k8s.io/node-stage-secret-name":             rbdNodeSecretName,
+				"csi.storage.k8s.io/controller-expand-secret-name":      rbdProvisionerSecretName,
+				"csi.storage.k8s.io/provisioner-secret-namespace":       consumerResource.Status.Client.OperatorNamespace,
+				"csi.storage.k8s.io/node-stage-secret-namespace":        consumerResource.Status.Client.OperatorNamespace,
+				"csi.storage.k8s.io/controller-expand-secret-namespace": consumerResource.Status.Client.OperatorNamespace,
+			}
+		} else if storageClassName == storagecluster.GenerateNameForCephFilesystemSC(storageCluster) {
+			provisioner = util.CephFSDriverName
+			annotations = map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			}
+			labels = map[string]string{
+				ramenDRStorageIDLabelKey: cephFsStorageID,
+			}
+			subVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
+			subVolumeGroup.Name = svgName
+			subVolumeGroup.Namespace = consumerResource.Namespace
+			err := s.client.Get(ctx, client.ObjectKeyFromObject(subVolumeGroup), subVolumeGroup)
+			if err != nil {
+				return nil, err
+			}
+			parameters = map[string]string{
+				"clusterID":          cephfsClientProfileName,
+				"subvolumegroupname": svgName,
+				"fsName":             subVolumeGroup.Spec.FilesystemName,
+				"pool":               subVolumeGroup.GetLabels()[ocsv1alpha1.CephFileSystemDataPoolLabel],
+				"csi.storage.k8s.io/provisioner-secret-name":            cephFsProvisionerSecretName,
+				"csi.storage.k8s.io/node-stage-secret-name":             cephFsNodeSecretName,
+				"csi.storage.k8s.io/controller-expand-secret-name":      cephFsProvisionerSecretName,
+				"csi.storage.k8s.io/provisioner-secret-namespace":       consumerResource.Status.Client.OperatorNamespace,
+				"csi.storage.k8s.io/node-stage-secret-namespace":        consumerResource.Status.Client.OperatorNamespace,
+				"csi.storage.k8s.io/controller-expand-secret-namespace": consumerResource.Status.Client.OperatorNamespace,
 			}
 		}
+		//TODO: Day-2 storageClasses
+
+		extR = append(extR,
+			&pb.ExternalResource{
+				Name: storageClassName,
+				Kind: "StorageClass",
+				Data: mustMarshal(&storagev1.StorageClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        storageClassName,
+						Annotations: annotations,
+						Labels:      labels,
+					},
+					ReclaimPolicy:        ptr.To(corev1.PersistentVolumeReclaimDelete),
+					AllowVolumeExpansion: ptr.To(true),
+					Provisioner:          provisioner,
+					Parameters:           parameters,
+				}),
+			},
+		)
 	}
 
-	if port < 0 {
-		return nil, fmt.Errorf("mgr pod port is empty")
+	for i := 0; i < len(consumerResource.Spec.VolumeSnapshotClasses); i++ {
```

Comment:
This if-else approach is unreasonable for so many built in *classes. Can we move to a template based approach where the template is chosen based on the name?